apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.Tests/ErrorReporting/ErrorReportingExceptionFilterTest.cs

```diff
@@ -0,0 +1,168 @@
+            mockContext.Setup(c => c.Request).Returns(mockRequest.Object);
+            mockContext.Setup(c => c.Response).Returns(mockResponse.Object);
+
+            Exception exception;
+            try
+            {
+                throw SimpleException;
+            }
+            catch (Exception e)
+            {
+                exception = e;
+            }
+
+            return new ExceptionContext
+            {
+                HttpContext = mockContext.Object,
+                Exception = exception
+            };
+        }
+
+        /// <summary>
+        /// Matcher to check if a <see cref="ReportedErrorEvent"/> matches one 
+        /// generated by <see cref="CreateComplexContext"/>
+        /// </summary>
+        private ReportedErrorEvent IsComplexContext()
+        {
+            bool isWindows = ErrorReportingUtils.IsWindows();
+            return Match.Create<ReportedErrorEvent>(e =>
+                e.Message.Contains(SimpleException.Message) &&
+                e.Context.HttpRequest.Method.Equals(DeleteMethod.Method) &&
+                e.Context.HttpRequest.Url.Equals(GoogleUri.ToString()) &&
+                e.Context.HttpRequest.UserAgent.Equals(UserAgent) &&
+                e.Context.HttpRequest.ResponseStatusCode == (int)ConflictStatusCode &&
+                (!isWindows || e.Context.ReportLocation.LineNumber > 0) &&
+                (!isWindows || !string.IsNullOrEmpty(e.Context.ReportLocation.FilePath)) &&
+                e.Context.ReportLocation.FunctionName.Equals(nameof(CreateComplexContext)) &&
+                e.ServiceContext.Service.Equals(ServiceName) &&
+                e.ServiceContext.Version.Equals(Version)
+            );
+        }
+
+        /// <summary>
+        /// Matcher to check if a <see cref="ReportedErrorEvent"/> matches
+        /// generated by <see cref="CreateSimpleContext"/>
+        /// </summary>
+        private ReportedErrorEvent IsSimpleContext()
+        {
+            return Match.Create<ReportedErrorEvent>(e =>
+                e.Message.Contains(SimpleException.Message) &&
+                string.IsNullOrEmpty(e.Context.HttpRequest.Method) &&
+                string.IsNullOrEmpty(e.Context.HttpRequest.Url) &&
+                string.IsNullOrEmpty(e.Context.HttpRequest.UserAgent) &&
+                e.Context.HttpRequest.ResponseStatusCode == 0 &&
+                e.Context.ReportLocation.LineNumber == 0 &&
+                string.IsNullOrEmpty(e.Context.ReportLocation.FilePath) &&
+                string.IsNullOrEmpty(e.Context.ReportLocation.FunctionName) &&
+                e.ServiceContext.Service.Equals(ServiceName) &&
+                e.ServiceContext.Version.Equals(Version)
+            );
+        }
+
+        private ErrorReportingExceptionFilter GetFilter(ReportErrorsServiceClient client)
+        {
+            return ErrorReportingExceptionFilter.Create(
+                Task.FromResult(client), ProjectId, ServiceName, Version);
```

Comment:
Why use `Task.FromResult` here? There's a Create overload which takes the client as-is (and yes, I realize that it in turn uses `Task.FromResult`, but that is an implementation detail of that class which could change in the future and the maintainer might not think to change this as well).